coordinator/performance/monitor.go

```diff
@@ -0,0 +1,79 @@
+package performance
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/coordinator/performance/prometheus"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+type Monitor struct {
+	config     MonitorConfig
+	helper     *prometheus.Helper
+	stopChan   chan struct{}
+	statusChan chan Status
+}
+
+func NewMonitor(config MonitorConfig) (*Monitor, error) {
+	if ok, err := config.IsValid(); !ok {
+		return nil, err
+	}
+	helper, err := prometheus.NewHelper(config.PrometheusURL)
+	if err != nil {
+		return nil, fmt.Errorf("performance: failed to create prometheus.Helper: %w", err)
+	}
+	return &Monitor{
+		config:     config,
+		helper:     helper,
+		stopChan:   make(chan struct{}),
+		statusChan: make(chan Status),
+	}, nil
+}
+
+func (m *Monitor) Run() (<-chan Status, error) {
```

Comment:
We are not returning any error anywhere. Do you foresee any future cases to return errors ?

Reply:
I guess running queries could result in some errors but will return those as part of the status so I think we can safely remove this.